Source/aweXpect.Reflection/ThatConstructor.cs

```diff
@@ -0,0 +1,314 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using aweXpect.Core;
+using aweXpect.Core.Constraints;
+using aweXpect.Options;
+using aweXpect.Reflection.Helpers;
+using aweXpect.Reflection.Options;
+using aweXpect.Results;
+
+namespace aweXpect.Reflection;
+
+/// <summary>
+///     Expectations on <see cref="ConstructorInfo" />.
+/// </summary>
+public static partial class ThatConstructor
+{
+	/// <summary>
+	///     Verifies that the <see cref="ConstructorInfo" /> has a parameter of type <typeparamref name="T" />.
+	/// </summary>
+	public static ConstructorHasParameter<T> HasParameter<T>(this IThat<ConstructorInfo?> subject)
```

Comment:
You should also add the corresponding expectations for the plural (with name "HaveParameter") both for an `IThat<IEnumerable<ConstructorInfo?>>` and for an `IThat<IEnumerable<MethodInfo?>>`.